src/kOS.Safe/Encapsulation/RangeValue.cs

```diff
@@ -0,0 +1,117 @@
+﻿using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Exceptions;
+using kOS.Safe.Serialization;
+using System;
+using System.Collections.Generic;
+
+namespace kOS.Safe
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Range")]
+    public class RangeValue : EnumerableValue<ScalarIntValue, Range>
+    {
+        private const string DumpFrom = "from";
+        private const string DumpTo = "to";
+        private const string DumpStep = "step";
+        private const string Label = "RANGE";
+
+        public static readonly int DEFAULT_FROM = 0;
+        public static readonly int DEFAULT_TO = 1;
+        public static readonly int DEFAULT_STEP = 1;
+
+        public RangeValue()
+            : this(DEFAULT_TO)
+        {
+        }
+
+        public RangeValue(int to)
+            : this(DEFAULT_FROM, to)
+        {
+        }
+
+        public RangeValue(int from, int to)
+            : this(from, to, DEFAULT_STEP)
+        {
+        }
+
+        public RangeValue(int from, int to, int step)
+            : base(Label, new Range(from, to, step))
+        {
+            InitializeRangeSuffixes();
+
+            if (step < 1)
+            {
+                throw new KOSException("Step must be a positive integer");
+            }
+        }
+
+        private void InitializeRangeSuffixes()
+        {
+            AddSuffix("FROM", new NoArgsSuffix<ScalarValue>(() => InnerEnumerable.From));
+            AddSuffix("TO", new NoArgsSuffix<ScalarValue>(() => InnerEnumerable.To));
```

Comment:
Ah crud - From/To the terminology isn't just in the docs.  It's also exposed here.  That means it bugs me a bit more than it otherwise would.  Docs can be changed or argued about post-release more easily than what the suffix is actually called can be.  Once we release with the exposed name in the suffix then we have to keep it that way or break scripts.